src/Storages/MergeTree/MergeTreeIndexText.cpp

```diff
@@ -399,49 +337,62 @@ DictionaryBlock deserializeDictionaryBlock(ReadBuffer & istr)
 
     std::vector<TokenPostingsInfo> token_infos;
     token_infos.reserve(num_tokens);
+    using enum PostingsSerialization::Flags;
 
     for (size_t i = 0; i < num_tokens; ++i)
     {
-        UInt64 header;
-        UInt32 cardinality;
+        TokenPostingsInfo info;
 
-        readVarUInt(header, istr);
-        readVarUInt(cardinality, istr);
+        readVarUInt(info.header, istr);
+        readVarUInt(info.cardinality, istr);
 
-        if (header & PostingsSerialization::EmbeddedPostings)
+        if (info.header & EmbeddedPostings)
         {
-            auto postings = PostingsSerialization::deserialize(header, cardinality, istr);
-            token_infos.emplace_back(std::move(postings));
+            auto postings = PostingsSerialization::deserialize(istr, info.header, info.cardinality);
+            info.offsets.emplace_back(0);
```

Comment:
In the case of embedded postings, why do we push an offset at all? Does an offset not only make sense for regular / non-embedded posting lists?